### PR TITLE
Fix heat transfer bugs

### PR DESCRIPTION
**Pull request by shahabgol**
# Description of the problem

- This PR fixes two bugs in the heat transfer solver:
1, source term is added to the strong residual vector
2, laser term is added to the local matrix of the Robin boundary condition.

# How Has This Been Tested?

- The results of one application test (on one and two CPUs) are updated.

### CODE REVIEW

File: source/solvers/heat_transfer_assemblers.cc [hunk 1]
```diff
@@ -164,8 +164,10 @@ HeatTransferAssemblerCore<dim>::assemble_rhs(
                       9 * std::pow(4 * alpha / (h * h), 2));
 
       // Calculate the strong residual for GLS stabilization
-      strong_residual_vec[q] += rho_cp * velocity * temperature_gradient -
-                                thermal_conductivity[q] * temperature_laplacian;
+      strong_residual_vec[q] +=
+        rho_cp * velocity * temperature_gradient -
+        thermal_conductivity[q] * temperature_laplacian +
+        scratch_data.source[q];
```
Comment (blaisb) [inline source/solvers/heat_transfer_assemblers.cc:170]: I think there is a sign error here. It should be - and not +.
The equation is rho_cp * u * grad(T) - k * laplacian T - source = 0
This is also what is coded in the RHS
k * (grad T) * grad(test) + rho_cp*u*grad(T) - source 

Only the laplacian changes sign due to integration by part.

**Review by blaisb — COMMENTED** [review]
Just a slight mistake to address.

File: source/solvers/heat_transfer_assemblers.cc (hunk 1 above)
Comment (shahabgol) [reply source/solvers/heat_transfer_assemblers.cc:170]: It is corrected. The errors also decreased in the failed application test. :)

**Comment by shahabgol** [comment]
@lpsaavedra Could you please review this PR and merge it if it's ok?

**Review by lpsaavedra — APPROVED** [review]
